daemon/main.c

```diff
@@ -532,6 +532,28 @@ main_terminate(
 //  Set process scheduling policy/priority and Lock resident memory.
 //
 
+static int move_to_root_cgroup(void) {
+  FILE *f;
+
+  f = fopen("/sys/fs/cgroup/cpu/tasks", "w");
+  if (f == NULL) {
+    log_message(LOG_DEBUG, "Can't open cgroups tasks file for writing\n");
+    return(-1);
+  }
+
+  if (fprintf(f, "%d\n", getpid()) <= 0) {
+		log_message(LOG_DEBUG, "Can't write pid into cgroups tasks file\n");
+    fclose(f);
+    return(-1);
+  }
+
+  if (fclose(f) != 0) {
+		log_message(LOG_DEBUG, "Can't close cgroups tasks file\n");
+    return(-1);
```

Comment:
If the return value is void then this will be redundant since it is the last statement in the function.